elide-async/src/main/java/com/yahoo/elide/async/service/thread/TableExportThread.java

```diff
@@ -157,4 +210,33 @@ protected String convertToCSV(PersistentResource resource) {
     protected void incrementRecordCount() {
         this.downloadRecordCount++;
     }
+
+    /**
+     * Generate CSV Header when Observable is Empty.
+     * @param parser TableExportParser object.
+     * @return returns Header string which is in CSV format.
+     */
+    protected Observable<String> generateCSVHeader(TableExportParser parser) {
+        EntityProjection projection = parser.parse(queryObj, apiVersion);
```

Comment:
Is this the 2nd time we construct the entity projection?  Ideally we should do this once.